### ISSUE TITLE
Adding an option to provide file extension, i.e. jsx, in the template

### ISSUE THREAD

**Issue by michaelrodov** [arminbro/generate-react-cli#25]
**Is your feature request related to a problem? Please describe.**
I like my jsx files to have an extension of jsx.
I couldnt find how to specify this in the template (named the component template "component.jsx" but that did nothing

**Describe the solution you'd like**
a config that would allow us to state the name of teh extension required

**Describe alternatives you've considered**
renaming manually

**Comment by boostup** [comment]
:+1: need this as well !!!

 `.jsx` extension adds loads of functionality in VSCode which are lost when using `.js` extension !

**Comment by arminbro** [comment]
@michaelrodov & @boostup  by resolving ticket https://github.com/arminbro/generate-react-cli/issues/19, this ticket should also now be resolved with the new release of [v5.1.0](https://github.com/arminbro/generate-react-cli/releases/tag/v5.1.0). 

GRC now supports custom extensions via custom templates. You should now be able to declare a custom component template with its own extension (`jsx`) within your `generate-react-cli.json` file. 

**Something like this:**
```json
{
  "usesTypeScript": false,
  "usesCssModule": true,
  "cssPreprocessor": "scss",
  "testLibrary": "Testing Library",
  "component": {
    "default": {
      "customTemplates": {
        "component": "templates/components/component.jsx"
      },
      "path": "src/components",
      "withStyle": true,
      "withTest": true,
      "withStory": true,
      "withLazy": false
    }
  }
}
```

Let me know if you guys run into any other problems. Thanks again for the request.
